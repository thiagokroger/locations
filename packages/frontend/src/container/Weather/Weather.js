import React from 'react'
import PropTypes from 'prop-types'
import moment from 'moment'

const Weather = ({ placeWeather }) => {
  return (
    <div>
      {
        placeWeather
          ? <div className="container weather-container">
            <div className="weather-header">
              <div className="column-title">{placeWeather.cityName}</div>
              {
                placeWeather.weatherDescriptions ? placeWeather.weatherDescriptions.map((weather, i) =>
                  <span key={i} className="weather-description">{weather}</span>)
                  : null}
              <div className="temperature-span">{placeWeather.temperature}째C</div>
            </div>
            <div className="weather-forecast columns is-multiline is-centered">
              {
                placeWeather.forecast
                  ? placeWeather.forecast.map((value, i) => (
                    <div key={i} className="column is-narrow">
                      <div className="weather-description">{moment(value.date).format('dddd, MMMM DD')}</div>

                      <div className="forecast-row">
                        <div className="forecast-label">Sunrise</div>
                        <div className="forecast-value">{value.sunrise}</div>
                      </div>
                      <div className="forecast-row">
                        <div className="forecast-label">Sunset</div>
                        <div className="forecast-value">{value.sunset}</div>
                      </div>
                      <div className="forecast-row">
                        <div className="forecast-label">Moonphase</div>
                        <div className="forecast-value">{value.moonPhase}</div>
                      </div>
                      <div className="forecast-row">
                        <div className="forecast-label">Min temp</div>
                        <div className="forecast-value">{value.mintemp}째C</div>
                      </div>
                      <div className="forecast-row">
                        <div className="forecast-label">Max temp</div>
                        <div className="forecast-value">{value.maxtemp}째C</div>
                      </div>
                      <div className="forecast-row">
                        <div className="forecast-label">Avg temp</div>
                        <div className="forecast-value">{value.avgtemp}째C</div>
                      </div>
                    </div>
                  ))
                  : null}

            </div>
          </div>
          : null
      }
    </div>
  )
}

Weather.propTypes = {
  placeWeather: PropTypes.any
}

export default Weather
