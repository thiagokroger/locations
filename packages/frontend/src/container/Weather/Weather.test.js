import React from 'react'
import { shallow } from 'enzyme'
import Weather from './Weather'

describe('Containers - Weather without placeWeather', () => {
  it('should render without crashing', () => {
    const component = shallow(
      <Weather
        placeWeather={null}
      />
    )
    expect(component).toMatchSnapshot()
  })
})

describe('Containers - Weather with placeWeather, without weatherDescriptions and forecast', () => {
  it('should render without crashing', () => {
    const component = shallow(
      <Weather
        placeWeather={{
          cityName: 'City Test',
          weatherDescriptions: null,
          forecast: null
        }}
      />
    )
    expect(component).toMatchSnapshot()
  })
})

describe('Containers - Weather with placeWeather, weatherDescriptions and forecast', () => {
  it('should render without crashing', () => {
    const component = shallow(
      <Weather
        placeWeather={{
          cityName: 'City Test',
          weatherDescriptions: ['Test Description'],
          forecast: [{
            date: 1587427200,
            sunrise: '06:10 AM',
            sunset: '06:10 PM',
            moonPhase: '',
            mintemp: 10,
            maxtemp: 20,
            avgtemp: 15
          }]
        }}
      />
    )
    expect(component).toMatchSnapshot()
  })
})
