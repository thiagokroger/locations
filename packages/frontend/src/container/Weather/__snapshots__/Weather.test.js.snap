// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Containers - Weather with placeWeather, weatherDescriptions and forecast should render without crashing 1`] = `
<div>
  <div
    className="container weather-container"
  >
    <div
      className="weather-header"
    >
      <div
        className="column-title"
      >
        City Test
      </div>
      <span
        className="weather-description"
        key="0"
      >
        Test Description
      </span>
      <div
        className="temperature-span"
      >
        °C
      </div>
    </div>
    <div
      className="weather-forecast columns is-multiline is-centered"
    >
      <div
        className="column is-narrow"
        key="0"
      >
        <div
          className="weather-description"
        >
          Monday, January 19
        </div>
        <div
          className="forecast-row"
        >
          <div
            className="forecast-label"
          >
            Sunrise
          </div>
          <div
            className="forecast-value"
          >
            06:10 AM
          </div>
        </div>
        <div
          className="forecast-row"
        >
          <div
            className="forecast-label"
          >
            Sunset
          </div>
          <div
            className="forecast-value"
          >
            06:10 PM
          </div>
        </div>
        <div
          className="forecast-row"
        >
          <div
            className="forecast-label"
          >
            Moonphase
          </div>
          <div
            className="forecast-value"
          />
        </div>
        <div
          className="forecast-row"
        >
          <div
            className="forecast-label"
          >
            Min temp
          </div>
          <div
            className="forecast-value"
          >
            10
            °C
          </div>
        </div>
        <div
          className="forecast-row"
        >
          <div
            className="forecast-label"
          >
            Max temp
          </div>
          <div
            className="forecast-value"
          >
            20
            °C
          </div>
        </div>
        <div
          className="forecast-row"
        >
          <div
            className="forecast-label"
          >
            Avg temp
          </div>
          <div
            className="forecast-value"
          >
            15
            °C
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Containers - Weather with placeWeather, without weatherDescriptions and forecast should render without crashing 1`] = `
<div>
  <div
    className="container weather-container"
  >
    <div
      className="weather-header"
    >
      <div
        className="column-title"
      >
        City Test
      </div>
      <div
        className="temperature-span"
      >
        °C
      </div>
    </div>
    <div
      className="weather-forecast columns is-multiline is-centered"
    />
  </div>
</div>
`;

exports[`Containers - Weather without placeWeather should render without crashing 1`] = `<div />`;
